src/app/(application)/layout.tsx

```diff
@@ -17,14 +25,15 @@ const AppLayout = ({ children }: { children: React.ReactNode }) => {
 
   return (
     <div>
-      {session?.user && isPremiumExpired && (
+      {session?.user && isPremiumExpired && isCommomUser && (
         <PremiumConversionModal
           isPremiumExpired={isPremiumExpired}
           isNewUser={session.user.subscriptionEndDate === null}
           setIsPremiumExpired={setIsPremiumExpired}
           userId={session.user.id}
         />
       )}
+      <CompleteSupplierRegister />
```

Comment:
We can make a rule to show this Modal here.

If supplier don't have a supplierLogo, Banner and description only.